src/SourcemapToolkit.SourcemapParser/SourceMap.cs

```diff
@@ -1,51 +1,133 @@
 
 namespace SourcemapToolkit.SourcemapParser
 {
-	public class SourceMap
-	{
-		/// <summary>
-		/// The version of the source map specification being used
-		/// </summary>
-		public int Version;
-
-		/// <summary>
-		/// The name of the generated file to which this source map corresponds
-		/// </summary>
-		public string File;
-
-		/// <summary>
-		/// The raw, unparsed mappings entry of the soure map
-		/// </summary>
-		public string Mappings;
-
-		/// <summary>
-		/// The list of source files that were the inputs used to generate this output file
-		/// </summary>
-		public List<string> Sources;
-
-		/// <summary>
-		/// A list of known original names for entries in this file
-		/// </summary>
-		public List<string> Names;
-
-		/// <summary>
-		/// Parsed version of the mappings string that is used for getting original names and source positions
-		/// </summary>
-		public List<MappingEntry> ParsedMappings;
-
-		/// <summary>
-		/// Finds the mapping entry for the generated source position. If no exact match is found, it will attempt 
-		/// to return a nearby mapping that should map to the same piece of code.
-		/// </summary>
-		/// <param name="generatedSourcePosition">The location in generated code for which we want to discover a mapping entry</param>
-		/// <returns>A mapping entry that is a close match for the desired generated code location</returns>
-		public virtual MappingEntry GetMappingEntryForGeneratedSourcePosition(SourcePosition generatedSourcePosition)
-		{
-			if (ParsedMappings == null)
-			{
-				return null;
-			}
+    public class SourceMap
+    {
+        /// <summary>
+        /// The version of the source map specification being used
+        /// </summary>
+        public int Version;
+
+        /// <summary>
+        /// The name of the generated file to which this source map corresponds
+        /// </summary>
+        public string File;
+
+        /// <summary>
+        /// The raw, unparsed mappings entry of the soure map
+        /// </summary>
+        public string Mappings;
+
+        /// <summary>
+        /// The list of source files that were the inputs used to generate this output file
+        /// </summary>
+        public List<string> Sources;
+
+        /// <summary>
+        /// A list of known original names for entries in this file
+        /// </summary>
+        public List<string> Names;
+
+        /// <summary>
+        /// Parsed version of the mappings string that is used for getting original names and source positions
+        /// </summary>
+        public List<MappingEntry> ParsedMappings;
+
+        /// <summary>
+        /// Applies the mappings of a sub source map to the current source map
+        /// Each mapping to the supplied source file is rewritten using the supplied source map
+        /// <param name="submap">The submap to apply</param>
+        /// <param name="aSourceFile">The filename of the source file. If not specified, submap's File property will be used</param>
```

Comment:
It's when the applied map is not in the same directory

example:
`a.js` has `sources: "../b.js"`
`b.map` has `File: "b.js"`

we would want to apply mappings in `a.js` with OriginalFileName "../b.js" instead of the submap's File attribute "b.js"